projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -57,7 +58,8 @@ export class NavigationUIComponent implements OnDestroy {
   constructor(
     private router: Router,
     private renderer: Renderer2,
-    private elemRef: ElementRef
+    private elemRef: ElementRef,
+    protected hamburgerMenuService?: HamburgerMenuService
```

Comment:
IMHO the generic navigation UI component should not know it's presented in Hamburger menu.

Currently only the `StorefrontComponent` is responsible for collapsing hamburger menu, so ideally we should keep the responsibility only there. 

Could you try to find a way to be notified in the `StorefrontComponent` that some navigation link was clicked in the header? One of the options to consider could be using `@ViewChildren` decorator.